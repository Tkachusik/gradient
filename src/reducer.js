const initialState = {
    redirectToHome: false,
    items: [],
}

const reducer = (state = initialState, action) => {
    switch (action.type) {
        case "CLICK_ADD":
            return {
                ...state,
                redirectToHome: true,
                items: [...state.items, action.item],
            }
        case "RESET_STATUS":
            return {
                ...state,
                redirectToHome: false,
            }
        case "DELETE_GRADIENT":
            return {
                ...state,
                items: state.items.filter((item, id) => id !== action.id),
            }
        case "CLICK_EDIT":
            return {
                ...state,
                items: state.items.map((item, idx) => {
                    if(idx.toString() === action.id) {
                        return action.item;
                    }

                    return item;
                }),
                redirectToHome: true,
            }
        default:
            return state;
    }
}
export default reducer;