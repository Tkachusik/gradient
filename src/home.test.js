import React from 'react';
import renderer from 'react-test-renderer';
import Home from './home';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import reducer from './reducer';
import { BrowserRouter as Router } from 'react-router-dom';

describe("Test home component", () => {
    test("Should render component", () => {
        const store = createStore(reducer);
        const component = renderer.create(
            <Provider store={store}>
                <Router>
                    <Home />
                </Router>
            </Provider>
        );
        let tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});