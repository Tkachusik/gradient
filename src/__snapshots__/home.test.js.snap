// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test home component Should render component 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row grid add-new-btn"
  >
    <a
      className="btn btn-success"
      href="/new"
      onClick={[Function]}
    >
      Add gradient
    </a>
  </div>
</div>
`;
